app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -0,0 +1,75 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.widget.Button;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnFocusChange;
+import butterknife.OnTextChanged;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+
+public class EntryNameActivity extends BaseMvpActivity<Presenter> implements View {
+
+    @BindView(R.id.contestant_welcome_text_view)
+    TextView contestantWelcomeTextView;
+    @BindView(R.id.entry_name_next_button)
+    Button entryNameNextButton;
+
+    @NonNull
+    @Override
+    public Presenter createPresenter(PresenterConfiguration configuration) {
+        return new EntryNamePresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_entry_name;
+    }
+
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+    }
+
+    @Override
+    public void onContestNameRetrieved(String contestName) {
+        contestantWelcomeTextView.setText(getResources().getString(R.string.contestant_welcome_message, contestName));
+    }
+
+    @OnFocusChange(R.id.entry_name_edit_text)
+    public void onEntryNameTextViewFocusChanged(boolean isFocused) {
+        presenter.onEntryNameTextViewFocusChanged(isFocused);
+    }
+
+    @OnTextChanged(R.id.entry_name_edit_text)
+    public void onEntryNameTextViewTextChanged(CharSequence newText, int start, int before, int count) {
+        presenter.onEntryNameTextViewTextChanged(newText);
+    }
+
+    @Override
+    public void showEntryNameNextButton() {
+        entryNameNextButton.setVisibility(android.view.View.VISIBLE);
```

Comment:
- The problem with the static import is that I'm already using `View` as the `EntryNameContract.View`:
``public class EntryNameActivity extends BaseMvpActivity<Presenter> implements View``
Do you think it's clearer if I leave `View` to refer to the `android.view.View`, and refer to the View interface as `EntryNameContract.View`?

- Simplify methods: I think I will wait until the designers decide if there will be animations, because depending on the code for that I think it might be better to leave it separate?